imap/sync_support.c

```diff
@@ -3072,6 +3083,27 @@ int sync_apply_mailbox(struct dlist *kin,
         goto done;
     }
 
+    /* take all userflags from the master */
+    if (dlist_isatomlist(userflags)) {
+        struct dlist *di;
+
+        for (di = userflags->head; di; di = di->next) {
+            const char *userflag;
+
+            if (di->type != DL_ATOM || !di->sval) continue; /* XXX ??? */
+
+            userflag = di->sval;
```

Comment:
why not use `dlist_toatom` here instead of inspecting inside?

Reply:
Yeah, didn't realise that one existed either!  Nearby code was just looking at sval directly, and I was uncomfortable enough while copying that to add a paranoid type check, but didn't see a better way.  Thanks for the tip! :)